### DISCUSSION TITLE
bySiteStructure with no Host directory

### DISCUSSION THREAD

**Discussion by well-made-uk** [website-scraper/node-website-scraper#548] [Q&A]
I am using bySiteStructure. This plugin creates a directory for the Host (e.g. outputFolder => www.example.com => index.html), which I don't want; I'd like the first index.html to be in the root of the output folder I specified. Can I change this behaviour?

**Comment by s0ph1e** [comment]
Hi @well-made-uk 👋 

Sorry for the late response. The host directory is created because the website may contain links to resources on other hosts. Without a host directory, resources with the same filename on different domains may be overwritten (e.g. if your website contains links to `https://site-a.com/img.png` and `https://site-b.com/img.png`)

In order to change this behaviour you [can write new plugin](https://github.com/website-scraper/node-website-scraper?tab=readme-ov-file#plugins) which implements [generateFilename action](https://github.com/website-scraper/node-website-scraper?tab=readme-ov-file#generatefilename). As an example you can use the code from [built-in bySiteStructure plugin](https://github.com/website-scraper/node-website-scraper/blob/da3c3c0ebb03dd11215ebeae4d6edbc41cd79a37/lib/filename-generator/by-site-structure.js) and make desired changes.
